server/internal/infrastructure/memory/item.go

```diff
@@ -0,0 +1,71 @@
+package memory
+
+import (
+	"context"
+
+	"github.com/reearth/reearth-cms/server/internal/usecase/repo"
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/item"
+	"github.com/reearth/reearthx/rerror"
+	"github.com/reearth/reearthx/util"
+	"golang.org/x/exp/slices"
+)
+
+type Item struct {
+	data *util.SyncMap[item.ID, *item.Item]
+	err  error
```

Comment:
```suggestion
	data *memorygit.VersionedSyncMap[item.ID, *item.Item]
```

use `memorygit.VersionedSyncMap`